provider/pkg/await/service.go

```diff
@@ -318,33 +321,25 @@ func (sia *serviceInitAwaiter) processEndpointEvent(event watch.Event, settledCh
 	inputServiceName := sia.config.currentOutputs.GetName()
 
 	// Get endpoint object.
-	endpoint, isUnstructured := event.Object.(*unstructured.Unstructured)
+	obj, isUnstructured := event.Object.(*unstructured.Unstructured)
 	if !isUnstructured {
 		logger.V(3).Infof("Endpoint watch received unknown object type %q",
-			reflect.TypeOf(endpoint))
+			reflect.TypeOf(obj))
 		return
 	}
 
 	// Ignore if it's not one of the endpoint objects created by the service.
 	//
 	// NOTE: Because the client pool is per-namespace, the endpointName can be used as an
 	// ID, as it's guaranteed by the API server to be unique.
-	if endpoint.GetName() != inputServiceName {
+	if obj.GetName() != inputServiceName {
 		return
 	}
 
-	// Start over, prove that service is ready.
-	sia.endpointsReady = false
-
-	// Update status of endpoint objects so we can check success.
-	if event.Type == watch.Added || event.Type == watch.Modified {
-		subsets, hasTargets := openapi.Pluck(endpoint.Object, "subsets")
-		targets, targetsIsSlice := subsets.([]any)
-		endpointTargetsPod := hasTargets && targetsIsSlice && len(targets) > 0
-
-		sia.endpointsReady = endpointTargetsPod
-	} else if event.Type == watch.Deleted {
-		sia.endpointsReady = false
+	var endpoints corev1.Endpoints
+	err := runtime.DefaultUnstructuredConverter.FromUnstructured(obj.Object, &endpoints)
```

Comment:
nit: do we need to attempt conversion from an unstructured to typed Endpoint object?

Reply:
I went this way since it made it easier to inspect the subsets, but I can understand the benefit to keeping everything as unstructured and version-agnostic as possible. In this situation the Informer is already using `v1/endpoints` so we shouldn't expect to see any other types, but I'm open to doing whatever you think is best.

Reply:
Ah! I didn't see `endpoints` being used further down (`for _, subset := range endpoints.Subsets {`) during the first pass, so I thought we were only just attempting to validate that the unstructured object could bind properly.